plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -378,9 +378,47 @@ def _units_jobs(self, units):
                 raise ValueError('Encountered an unknown unit {}'.format(unit))
             yield self.pool.Job(flags, solvable.id)
 
+    def missing_dependencies_to_solvables(self, problems):
+        # Convert missing dependency problems into new solvables to be able to retry the solving
+        # by pretending the missing dependencies are already installed
+        if not self.target_repo:
+            return
+        target_repo = self.mapping.repos[self.target_repo.repo_id]
+        for problem in problems:
+            for problem_rule in problem.findallproblemrules():
+                _LOGGER.warning('problem rule #%s', problem_rule)
+                for info in problem_rule.allinfos():
+                    name = info.dep.str()
+                    if info.type == solv.Solver.SOLVER_RULE_PKG_NOTHING_PROVIDES_DEP:
+                        dummy = target_repo.add_solvable()
+                        dummy.add_deparray(solv.SOLVABLE_PROVIDES, info.dep)
+                        _LOGGER.warning('Created dummy: %s', name)
+                    else:
+                        _LOGGER.warning('Got no workaround for solver problem rule %(p)s type %(t)s ',
+                        {'p': name, 't': info.type})
+        self.pool.createwhatprovides()
+
     def find_dependent_rpms(self, units):
-        solver = self.pool.Solver()
-        problems = solver.solve(self._units_jobs(units))
+        ret = set()
+        if not units:
+            return ret
+        previous_problem_ids = set()
+        take = 0
+        while True:
+            take += 1
+            _LOGGER.warning('Solver take %(t)s; units %(u)s', {'t': take, 'u': units})
+            solver = self.pool.Solver()
+            problems = solver.solve(self._units_jobs(units))
+            problem_ids = {str(problem) for problem in problems}
+            # assuming we won't be creating more and more problems all the time
+            _LOGGER.warning('previous problems #%s', previous_problem_ids)
```

Comment:
I think warning might be too loud. Perhaps it should be at INFO or DEBUG.

Reply:
@dkliban yup will update with a saner logging; current revision is just to have a base for the discussion